Source/WebCore/page/FrameTree.cpp

```diff
@@ -537,6 +520,22 @@ unsigned FrameTree::depth() const
     return depth;
 }
 
+AtomString FrameTree::uniqueName() const
+{
+    if (!parent())
+        return m_specifiedName;
+
+    auto frameIndex { 0u };
+    for (RefPtr frame = top().tree().firstChild(); frame; frame = frame->tree().traverseNext()) {
+        auto frameName = frame->tree().specifiedName();
+        if (frame->tree().specifiedName().isEmpty() || isBlankTargetFrameName(frameName) || frame->tree().childBySpecifiedName(frameName))
+            frameName = makeAtomString("<!--frame"_s, ++frameIndex, "-->"_s);
```

Comment:
This makes a lot of unnecessary AtomStrings.  Couldn't we just increment frameIndex here and only create one AtomString as we return?

Reply:
Yes. Fixed.